src/Agent.Listener/CommandSettings.cs

```diff
@@ -251,13 +254,19 @@ public string GetToken()
                 validator: Validators.NonEmptyValidator);
         }
 
-        // TODO: REMOVE defaultValue parameter after fix in master and rebase on master
-        public string GetUrl(string defaultValue = null)
+        public string GetUrl(bool suppressPrompt = false)
```

Comment:
the problem i'm trying to solve here is: the url is sometimes optional for localRun. If the url is specified, we use it and download newer tasks if available. otherwise we lazily prompt if a task is referenced that doesnt exist in the cache.